pytorch_lightning/core/lightning.py

```diff
@@ -68,6 +68,21 @@ def __init__(self, *args, **kwargs):
         #: True if using amp
         self.use_amp = False
 
+    @property
+    def hparams(self) -> Namespace:
+        if not hasattr(self, '_hparams'):
+            self._hparams = Namespace()
+        if isinstance(self._hparams, Namespace):
+            hparams = self._hparams
+        else:
+            assert isinstance(self._hparams, dict)
+            hparams = Namespace(**self._hparams)
+        return hparams
+
+    def set_hparams(self, params: Union[Dict[str, Any], Namespace]):
```

Comment:
setters in python should be done with `@hparams.setter` decorator, no?

Reply:
can you in decorator define types?

Reply:
```
@hparams.setter
def hparams(self, params: Union[Dict[str, Any], Namespace]):
    self._hparams = params
```
This should work. However, it does not enforce the types. The user can still pass in whatever it wants.